### PR TITLE
refactor unit tests, replace ava with tap, remove undici and typescript as devDependencies

### PR DESCRIPTION
**Pull request by Uzlopak**
As first step to get this module improved, I refactored the unit tests by replacing ava and undici with tap. So unit tests got blazingly fast and by using fastify.inject it works now the "fastify-way". 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,20 +31,17 @@
...
+    "fastify": "^4.3.0",
     "fastify-cookie": "~5.7.0",
     "ioredis": "^5.0.5",
-    "nyc": "^15.0.0",
     "published-session": "npm:@fastify/session@latest",
     "session-file-store": "^1.5.0",
     "sinon": "^14.0.0",
     "standard": "^17.0.0",
-    "tsd": "^0.22.0",
-    "typescript": "^4.0.2",
```
Comment (kibertoad) [inline package.json:46]: typescript wasn't used for anything?

Comment (Uzlopak) [reply package.json:46]: typescript is anyway a dependency of tsd.

Comment (Uzlopak) [reply package.json:46]: well actually tsd uses its own fork. So having typescript as dependency is basically pointless if you think you need it for tsd

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**